hardware-computing/README.md

```diff
@@ -1,11 +1,95 @@
+The hardware implementation is much harder to be affected by such attacks.
+
+Intel, AMD, ARM, Risc-V SPARC, IBM have implemented AES instruction sets in hardware.
+We will discuss the implentations from Intel and ARM.
+If interested in the implementations from other architectures, you can find them in the `Further Reading` section.
+
+If you are not familiar with how AES works, you can find details about it in the `helpers/` folder, in the `aes.md` file.
+
+### Intel AES-NI
+
+In 2008, Intel proposed an AES instruction set for the x86_64 CPUs, AES-NI (Advanced Encryption Standard New Instructions).
+AMD also implemented the instruction set for their CPUs.
+
+The instruction set provides 2 instructions for encrypting, 2 for decrypting, and 2 for key generation:
+ * AESENC performs an encryption round
+ * AESENCLAST performs the last encryption round
+ * AESDEC performs a decryption round
+ * AESDECLAST performs the last decryption round
+ * AESKEYGENASSIS aids in the key expansion phase of the algorithm
+ * AEIMC prepares a key for being used in decryption
+
+All the instructions use XMM registers, which are part of the SSE / AVX instruction sets.
+If you are unsure what those are, you can find information about them in the `helpers/` folder, in the `simd-instructions.md` file.
+
+In the case of the first 4 and the 6th instructions, 2 XMM registers are used: the first one stores the intermediary state of the cyphertext, and the second one the key used in that round.
+The 5th one also takes an immediate value, called RCON, which is used in the key generation algorithm.
+
+For those of you that don't like the Assembly language, there are good news.
+The AES-NI instructions can be also used in C, not only in Assembly, with the so-called **intrinsics**.
+
+The tasks for this part will be straight-forward:
+
+#### 1. Challenge: AES in C
+
+You are provided with an implementation of AES encryption, in C.
+Complete the code with the decryption implementation, obtain the flag and submit it.
+
+#### 2. Challenge: AES in ASM
+
+You are provided with an implementation of AES encryption, in Assembly.
+Complete the code with the decryption implementation, obtain the flag and submit it.
+
+### ARM Cryptography Extensions
+
+While Intel and AMD provide the AES-NI instruction set with all their newer processors, ARM provides such instructions as an optional extension module, called the "Cryptography Extension".
+This module is avaible with the ARMv8 base ISA.
+It doesn't support all operations associated with AES encryption and decryption, but provides some instructions for the SHA hashing algorithm.
+For AES, the following instructions are provided:
+ * AESE, which performs a single encryption round
+ * AESD, which performs a single decryption round
+ * AESMC, which performs the `mix collumns` operation
+ * AESIMC, which performs the `inverse mix collumns` operation
```

Comment:
```suggestion
 * AESMC, which performs the `MixColumns` operation
 * AESIMC, which performs the `InverseMixColumns` operation
```

So that they're consistent with Rijndael's notations.